picard/util/__init__.py

```diff
@@ -541,9 +541,19 @@ def linear_combination_of_weights(parts):
 
 
 def album_artist_from_path(filename, album, artist):
-    """If album is not set, try to extract album and artist from path
+    """If album is not set, try to extract album and artist from path.
+
+    Args:
+        filename: The full file path
+        album: Default album name
+        artist: Default artist name
+
+    Returns:
+        A tuple (album, artist)
     """
     if not album:
+        if IS_WIN:
+            filename = ntpath.splitdrive(filename)[1]
```

Comment:
We can even get rid of this using `pathlib.PurePath`

https://github.com/phw/picard/pull/13